cmd/commands/cmd_walletunlocker.go

```diff
@@ -366,6 +376,23 @@ mnemonicCheck:
 		printCipherSeedWords(cipherSeedMnemonic)
 	}
 
+	// Parse the macaroon root key if it was specified by the user.
+	if ctx.IsSet(macRootKeyFlag.Name) {
+		macRootKey, err = hex.DecodeString(
+			ctx.String(macRootKeyFlag.Name),
+		)
+		if err != nil {
+			return fmt.Errorf("unable to parse macaroon root key: "+
+				"%w", err)
+		}
+
+		if len(macRootKey) != macaroons.RootKeyLen {
+			return fmt.Errorf("macaroon root key must be exactly "+
+				"%v bytes, got %v", macaroons.RootKeyLen,
+				len(macRootKey))
+		}
```

Comment:
this is checked on the server side (`(u *UnlockerService) InitWallet`) so maybe not needed here? but no harm in checking before sending i guess!

Reply:
Yeah, I guess no harm in avoiding a round trip to the server if the data isn't valid in the first place. Also, the error is likely a bit more easy to understand (`[lncli] macaroon root key ...` vs. `[lncli] [lncli] rpc error: code = Unknown desc = macaroon root key ...`.